rust/perspective-server/build/psp.rs

```diff
@@ -26,10 +26,24 @@ pub fn cmake_build() -> Result<Option<PathBuf>, std::io::Error> {
     let profile = std::env::var("PROFILE").unwrap();
     dst.always_configure(true);
     dst.define("CMAKE_BUILD_TYPE", profile.as_str());
-    if std::env::var("PSP_ARCH").as_deref() == Ok("x86_64") {
-        dst.define("CMAKE_OSX_ARCHITECTURES", "x86_64");
-    } else if std::env::var("PSP_ARCH").as_deref() == Ok("arm64") {
-        dst.define("CMAKE_OSX_ARCHITECTURES", "arm64");
+
+    if cfg!(target_os = "macos") {
+        // Set CMAKE_OSX_ARCHITECTURES et al. for Mac builds.  Arrow does not forward on
+        // CMAKE_OSX_ARCHITECTURES and but it does forward on a
+        // CMAKE_TOOLCHAIN_FILE. In Conda builds, the environment sets
+        // `CMAKE_ARGS` up with various toolchain arguments. This block may need
+        // to be patched out or adjusted for Conda.
```

Comment:
by this I mean this block may need some `if !conda_build` style guards. I am not sure which wins out between a value set in `CMAKE_TOOLCHAIN_FILE` vs. a value set later at the command line

Reply:
funny enough, the conda 3.1.1 macos cross-compiling builds were apparently unaffected -- their import tests succeeded. https://github.com/conda-forge/perspective-feedstock/runs/31853180964

I'm not exactly sure why, but conda-build passes its toolchain arguments to us (in CMAKE_ARGS) with compilers/tools with arm64 in the name, `-DCMAKE_C_COMPILER=arm64-…-cc -DCMAKE_LIBTOOL=arm64-…-cc`, which I presume is a compiler toolchain which targets arm64 without needing any other flags, and doesn't set `CMAKE_OSX_ARCHITECTURE` itself. And Arrow does forward those specific flags on to its external dependencies